samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -241,6 +241,11 @@ public void requestResources(SamzaResourceRequest resourceRequest) {
     String processorId = resourceRequest.getProcessorId();
     String requestId = resourceRequest.getRequestId();
     String preferredHost = resourceRequest.getPreferredHost();
+    String[] racks = null;
+    ClusterManagerConfig clusterManagerConfig = new ClusterManagerConfig(config);
+    if (clusterManagerConfig.getFaultDomainAwareStandbyEnabled()) {
+      racks = resourceRequest.getFaultDomains().stream().map(FaultDomain::getId).toArray(String[]::new);
+    }
```

Comment:
What do you think about not having control flow here and pass empty array instead of null if it doesn't change things semantically in YARN?

By doing so, you will just eliminate unnecessary control flow and given faultDomains is guaranteed to be empty or present, null handling is not necessary as well.

Reply:
Sure, that makes sense. I've removed the unnecessary check.